workbench/tests/renderer/plugin.test.js

```diff
@@ -73,13 +73,14 @@ describe('Add plugin modal', () => {
     const submitButton = await findByText('Add');
     userEvent.click(submitButton);
 
-    await findByText('Loading...');
-    const calledChannels = spy.mock.calls.map((call) => call[0]);
+    await findByText('Adding...');
     await waitFor(() => {
+      const calledChannels = spy.mock.calls.map((call) => call[0]);
       expect(calledChannels).toContain(ipcMainChannels.ADD_PLUGIN);
     });
-    // expect the plugin dialog to have disappeared
-    await waitFor(() => expect(queryByRole('dialog')).toBeNull());
+    // close the modal
+    const overlay = await findByRole('dialog');
+    await act(async () => { await userEvent.click(overlay); });
```

Comment:
From the look of it, this should work just the same as `await userEvent.click(overlay);`

It looks like there are a few other cases in this file where `userEvent` is sometimes `awaited`, sometimes not, and sometimes wrapped in `act`. I think the cleanest and most reliable pattern is to always do,

`await userEvent.*`

Reply:
I was attempting to resolve the `Warning: An update to Transition inside a test was not wrapped in act(...).` messages, but you're right this doesn't seem to be doing anything here. I added `await` to all the `userEvent` calls in the test suite, and that removes the `act()` warnings... even when the warning is coming from a different line of code.

Reply:
Great! I think that's the right approach. My understanding is that even though React issues those `not wrapped in act` warnings, the correct approach for us, since we're using `testing-library` APIs, is basically never to wrap in `act`, but to look for something async that we forgot to `await`.